core/src/sql/number.rs

```diff
@@ -589,10 +591,27 @@ impl Ord for Number {
 			(Number::Decimal(v), Number::Int(w)) => v.cmp(&Decimal::from(*w)),
 			// ------------------------------
 			(Number::Float(v), Number::Decimal(w)) => {
-				// `rust_decimal::Decimal` code comments indicate that `to_f64` is infallible
-				total_cmp_f64(*v, w.to_f64().unwrap())
+				if !v.is_finite() {
+					v.total_cmp(&0.0)
+				} else if *v >= MAX_DECIMAL_IN_F64 {
+					Ordering::Greater
+				} else if *v <= -MAX_DECIMAL_IN_F64 {
+					Ordering::Less
+				} else {
+					Decimal::cmp(&Decimal::from_f64(*v).expect("checked that float is finite"), &w)
+				}
```

Comment:
Instead of checking for infinity separately and hard-coding the max decimal, have you considered this?
```rust
if let Some(v) = Decimal::from_f64_retain(*v) {
    Decimal::cmp(&v, &w)
} else {
    v.total_cmp(&0.0) // -inf, inf, NaN's, overflows
}
```

(also note [`from_f64_retain`](https://docs.rs/rust_decimal/latest/rust_decimal/struct.Decimal.html#method.from_f64_retain) is probably more appropriate in this case).

Reply:
This implementation is not correct, as overflows don't return none, this code runs:
```rust
let float = 79228162514264337593543950336f64;
let dec = Decimal::from_str_exact("79228162514264337593543950330").unwrap();
assert_eq!(Decimal::from_f64(float).unwrap(), dec);
```

Reply:
@RaphaelDarley I saw your deleted comment! While my implementation seems correct, you did find a bug in `rust_decimal` that affects just two inputs.

I've fixed it here: https://github.com/paupino/rust-decimal/pull/678

For now, if `v.abs() == 79228162514264337593543950336`, then can act like it overflowed:
```rust
if let Some(v) = Decimal::from_f64_retain(*v).filter(|_| v.abs() != 79228162514264337593543950336) {
    Decimal::cmp(&v, &w) // in range
} else if v.is_sign_positive() {
    Ordering::Greater // inf, +NaN, pos overflow
} else {
    Ordering::Less // -inf, -NaN, neg overflow
}
```

Reply:
Thanks, I'll have a look, I'm also having some issues with subnormal floats, but hopefully it should come together soon 🤞